_source/_posts/2021-02-26-unity-webgl-playfab-authorization.md

```diff
@@ -0,0 +1,654 @@
+
+{% img blog/unity-webgl-playfab-authorization/8_popup.png alt:"Popup" width:"800" %}{: .center-image }
+
+Ok, great. But how does this relate to the build target? The browser. OAuth relies on a browser to facilitate user authentication, which must validate the user identity before providing authorization. Every platform will have a different, platform-specific way of presenting the user with a browser to interact with. For WebGL, the game is already running in a browser, so a simple pop up is all that is needed. But if the build target were Android, [Chrome Custom Tabs](https://developer.chrome.com/docs/multidevice/android/customtabs/) would instead be needed to embed the browser into the app. Similarly, if the build target was iOS, [Safari View Controller](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller) would be needed to embed the browser into the app. For full-screen games on a PC, it would be better to use a [device code](https://oauth.net/2/grant-types/device-code/) concept, similar to a TV or IoT device, or even authenticate the player before launching the game client; from an external launcher application interacting with the default browser on the operating system. The browser interaction and design will change for every platform and device. When designing the authentication experience, the build target will heavily influence the rest of the design.
+
+The good news is that once the target is defined and the browser interaction decided, the rest of the code should be similar across all platforms. This is thanks to the shared responsibility with a security platform's hosted login experience and a standardized authorization specification. The job of the developer is to securely provide a platform-preferred browser experience for the user and direct them to the OAuth authorize endpoint. Once the user has authenticated, the browser is closed, and an auth code is captured in client code to be exchanged for tokens behind the scenes. With tokens in hand, user authorization can be passed to all other backend cloud platforms; providing a centralized, and secure, identity security design that drastically simplifies the implementation effort.
+
+## Create the Project
+
+First, if you have not already done so, create a new project from `Unity Hub` with the `3D` template.
```

Comment:
I suggest leading into this with a sentence to the effect of "To see how all of this would look in practice, let's take a look at how to do this with code in Unity"

Reply:
Got it.